test/functionalTest/cases/722_unhardwire_format_cpr/notify_format_in_ngsi9_subscriptions.test

```diff
@@ -0,0 +1,944 @@
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart
+
+--SHELL--
+
+#
+# The format for Subscriptions are sent in either JSON or XML, depending on two things:
+# 1. The URI parameter notifyFormat
+# 2. The format that the subscription in sent in
+#
+# The URI parameter notifyFormat (1) overrides format that the subscription in sent in (2).
+#
+# So, we have a number of combinations here:
+# 1. Create a subscription in JSON, without using URI parameter notifyFormat - see notification in JSON
+# 2. Create a subscription in XML, without using URI parameter notifyFormat - see notification in XML
+# 3. Create a subscription in JSON, with URI parameter notifyFormat in JSON - see notification in JSON
+# 4. Create a subscription in JSON, with URI parameter notifyFormat in XML - see notification in XML
+# 5. Create a subscription in XML, with URI parameter notifyFormat in JSON - see notification in JSON
+# 6. Create a subscription in XML, with URI parameter notifyFormat in XML - see notification in XML
+#
+# 
+# TEST PLAN:
+# 0.1. Availability-Subscribe to E1/A0 in JSON, with invalid notifyFormat=ABCD
+# 0.2. Availability-Subscribe to E1/A0 in XML, with invalid notifyFormat=ABCD
+#
+# 1.1. Register E1/T1/A1
+# 1.2. Availability-Subscribe to E1/A1 in JSON 
+# 1.3. Dump accumulator to see notification of A1 in JSON, as the subscription was done in JSON
+# 1.4. Reset the accumulator
+#
+#   1.5. Modify subscription in XML (later)
+#   1.6. Dump accumulator to see notification of A1 in XML, as the subscription modification was done in XML (later
+#   1.7. Reset the accumulator (later)
+#   1.8. Modify subscription in XML, with URI parameter notifyFormat=JSON (later)
+#   1.9. Dump accumulator to see notification of A1 in JSON, as the subscription modification was done with notifyFormat=JSON (later)
+# 
+# 2.1. Register E1/T1/A2
+# 2.2. Availability-Subscribe to E1/A2 in XML
+# 2.3. Dump accumulator to see notification of A2 in XML, as the subscription was done in XML
+# 2.4. Reset the accumulator
+#
+# 3.1. Register E1/T1/A3
+# 3.2. Availability-Subscribe to E1/A3 in JSON, with URI parameter notifyFormat=JSON
+# 3.3. Dump accumulator to see notification of A3 in JSON, as URI parameter notifyFormat was JSON
+# 3.4. Reset the accumulator
+#
+# 4.1. Register E1/T1/A4
+# 4.2. Availability-Subscribe to E1/A4 in JSON, with URI parameter notifyFormat=XML
+# 4.3. Dump accumulator to see notification of A4 in XML, as URI parameter notifyFormat was XML
+# 4.4. Reset the accumulator
+#
+# 5.1. Register E1/T1/A5
+# 5.2. Availability-Subscribe to E1/A5 in XML, with URI parameter notifyFormat=JSON
+# 5.3. Dump accumulator to see notification of A5 in JSON, as URI parameter notifyFormat was JSON
+# 5.4. Reset the accumulator
+#
+# 6.1. Register E1/T1/A6
+# 6.2. Availability-Subscribe to E1/A6 in XML, with URI parameter notifyFormat=XML
+# 6.3. Dump accumulator to see notification of A6 in XML, as URI parameter notifyFormat was XML
+# 6.4. Reset the accumulator
+#
```

Comment:
However, on a second thought, maybe is better to merge this PR without the test, then do the timeout thing, then if we have still time in the sprint, complete with the NGSI10 subscription test.

Maybe we should create an issue for not forgetting this? Or would it be excessive?